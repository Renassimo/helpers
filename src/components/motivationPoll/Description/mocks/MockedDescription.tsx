const MockedDescription = jest.fn(() => <div>Mocked Description</div>);

export default MockedDescription;
