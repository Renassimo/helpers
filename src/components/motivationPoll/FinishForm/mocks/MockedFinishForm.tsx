const MockedFinishForm = jest.fn(() => <div>Mocked Finish Form</div>);

export default MockedFinishForm;
