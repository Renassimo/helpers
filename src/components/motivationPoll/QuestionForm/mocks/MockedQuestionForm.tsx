const MockedQuestionForm = jest.fn(() => <div>Mocked Question Form</div>);

export default MockedQuestionForm;
