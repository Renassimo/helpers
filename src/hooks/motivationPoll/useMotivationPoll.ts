import { useContext } from 'react';

import MotivationPollContext from 'src/contexts/motivationPoll';

const useMotivationPoll = () => useContext(MotivationPollContext);

export default useMotivationPoll;
