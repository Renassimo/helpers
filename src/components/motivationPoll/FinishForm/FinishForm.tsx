import { ChangeEvent, useCallback } from 'react';

import useMotivationPoll from '@/hooks/motivationPoll/useMotivationPoll';

import TextField from '@mui/material/TextField';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';

const FinishForm = () => {
  const { name, setName, prepareResults } = useMotivationPoll();

  const onNameChange = useCallback(
    (event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) =>
      setName(event.target.value),
    [setName]
  );

  return (
    <Box
      textAlign="center"
      display="flex"
      flexDirection="column"
      alignItems="center"
    >
      <Typography component="h1" variant="h5" textAlign="center" m={2}>
        Type you name an get your results
      </Typography>
      <Box
        maxWidth="500px"
        display="grid"
        gridTemplateColumns="1fr 1fr"
        gap="8px"
      >
        <TextField name="name" value={name || ''} onChange={onNameChange} />
        <Button
          disabled={!name}
          type="button"
          variant="outlined"
          size="large"
          onClick={prepareResults}
        >
          Get Results
        </Button>
      </Box>
    </Box>
  );
};

export default FinishForm;
