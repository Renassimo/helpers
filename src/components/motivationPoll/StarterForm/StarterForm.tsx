import useMotivationPoll from '@/hooks/motivationPoll/useMotivationPoll';

import Box from '@mui/material/Box';
import Button from '@mui/material/Button';

const StarterForm = () => {
  const { startTest } = useMotivationPoll();
  return (
    <Box textAlign="center">
      <Button type="button" variant="outlined" size="large" onClick={startTest}>
        Start
      </Button>
    </Box>
  );
};

export default StarterForm;
