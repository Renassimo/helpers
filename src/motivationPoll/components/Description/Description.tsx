import useMotivationPoll from '@/hooks/motivationPoll/useMotivationPoll';

import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';

const Description = () => {
  const { description } = useMotivationPoll();
  return (
    <Box>
      <Typography component="h1" variant="h4" textAlign="center" m={2}>
        Motivation Test
      </Typography>
      <Typography component="p" m={4}>
        {description}
      </Typography>
    </Box>
  );
};

export default Description;
