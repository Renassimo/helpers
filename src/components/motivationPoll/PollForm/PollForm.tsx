import useMotivationPoll from '@/hooks/motivationPoll/useMotivationPoll';

import Description from '@/components/motivationPoll/Description';
import StarterForm from '@/components/motivationPoll/StarterForm';
import QuestionForm from '@/components/motivationPoll/QuestionForm';
import FinishForm from '@/components/motivationPoll/FinishForm';

import Box from '@mui/material/Box';

const PollForm = () => {
  const { isStarted, isFinished, currentQuestion } = useMotivationPoll();

  return (
    <Box>
      {!isFinished && <Description />}
      {!isStarted && <StarterForm />}
      {currentQuestion && <QuestionForm />}
      {isFinished && <FinishForm />}
    </Box>
  );
};

export default PollForm;
