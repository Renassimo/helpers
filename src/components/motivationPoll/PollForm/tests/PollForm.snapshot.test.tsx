import renderWithTheme from '@/tests/helpers';

import useMotivationPoll from '@/hooks/motivationPoll/useMotivationPoll';

import Description from '@/components/motivationPoll/Description';
import StarterForm from '@/components/motivationPoll/StarterForm';
import QuestionForm from '@/components/motivationPoll/QuestionForm';
import FinishForm from '@/components/motivationPoll/FinishForm';

import MockedDescription from '@/components/motivationPoll/Description/mocks';
import MockedStarterForm from '@/components/motivationPoll/StarterForm/mocks';
import MockedQuestionForm from '@/components/motivationPoll/QuestionForm/mocks';
import MockedFinishForm from '@/components/motivationPoll/FinishForm/mocks';
import PollForm from '../PollForm';

jest.mock('@/hooks/motivationPoll/useMotivationPoll');
jest.mock('@/components/motivationPoll/StarterForm');
jest.mock('@/components/motivationPoll/Description');
jest.mock('@/components/motivationPoll/QuestionForm');
jest.mock('@/components/motivationPoll/FinishForm');

describe('PollForm snapshot', () => {
  beforeEach(() => {
    (useMotivationPoll as jest.Mock).mockImplementation(() => ({
      isFinished: true,
      isStarted: true,
    }));
    (Description as jest.Mock).mockImplementation(MockedDescription);
    (StarterForm as jest.Mock).mockImplementation(MockedStarterForm);
    (QuestionForm as jest.Mock).mockImplementation(MockedQuestionForm);
    (FinishForm as jest.Mock).mockImplementation(MockedFinishForm);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders FinishFrom', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(<PollForm />);
    // Assert
    expect(baseElement).toMatchSnapshot();
  });
  describe('when poll is not finished', () => {
    beforeEach(() => {
      (useMotivationPoll as jest.Mock).mockImplementation(() => ({
        isFinished: false,
        isStarted: true,
      }));
    });

    test('renders with description', () => {
      // Arange
      // Act
      const { baseElement } = renderWithTheme(<PollForm />);
      // Assert
      expect(baseElement).toMatchSnapshot();
    });
    describe('and when pol is not started', () => {
      beforeEach(() => {
        (useMotivationPoll as jest.Mock).mockImplementation(() => ({
          isFinished: false,
          isStarted: false,
        }));
      });
      test('renders with StarterForm', () => {
        // Arange
        // Act
        const { baseElement } = renderWithTheme(<PollForm />);
        // Assert
        expect(baseElement).toMatchSnapshot();
      });
      describe('and when has current question', () => {
        beforeEach(() => {
          (useMotivationPoll as jest.Mock).mockImplementation(() => ({
            isFinished: false,
            isStarted: false,
            currentQuestion: 'Current question?',
          }));
        });
        test('renders with QuestionForm', () => {
          // Arange
          // Act
          const { baseElement } = renderWithTheme(<PollForm />);
          // Assert
          expect(baseElement).toMatchSnapshot();
        });
      });
    });
  });
  describe('when pol is not started', () => {
    beforeEach(() => {
      (useMotivationPoll as jest.Mock).mockImplementation(() => ({
        isFinished: true,
        isStarted: false,
      }));
    });
    test('renders with StarterForm', () => {
      // Arange
      // Act
      const { baseElement } = renderWithTheme(<PollForm />);
      // Assert
      expect(baseElement).toMatchSnapshot();
    });
    describe('and when has current question', () => {
      beforeEach(() => {
        (useMotivationPoll as jest.Mock).mockImplementation(() => ({
          isFinished: true,
          isStarted: false,
          currentQuestion: 'Current question?',
        }));
      });
      test('renders with QuestionForm', () => {
        // Arange
        // Act
        const { baseElement } = renderWithTheme(<PollForm />);
        // Assert
        expect(baseElement).toMatchSnapshot();
      });
    });
  });
});
