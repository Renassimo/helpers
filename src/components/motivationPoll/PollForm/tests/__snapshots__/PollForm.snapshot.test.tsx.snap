// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollForm snapshot renders FinishFrom 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div>
        Mocked Finish Form
      </div>
    </div>
  </div>
</body>
`;

exports[`PollForm snapshot when pol is not started and when has current question renders with QuestionForm 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div>
        Mocked Starter Form
      </div>
      <div>
        Mocked Question Form
      </div>
      <div>
        Mocked Finish Form
      </div>
    </div>
  </div>
</body>
`;

exports[`PollForm snapshot when pol is not started renders with StarterForm 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div>
        Mocked Starter Form
      </div>
      <div>
        Mocked Finish Form
      </div>
    </div>
  </div>
</body>
`;

exports[`PollForm snapshot when poll is not finished and when pol is not started and when has current question renders with QuestionForm 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div>
        Mocked Description
      </div>
      <div>
        Mocked Starter Form
      </div>
      <div>
        Mocked Question Form
      </div>
    </div>
  </div>
</body>
`;

exports[`PollForm snapshot when poll is not finished and when pol is not started renders with StarterForm 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div>
        Mocked Description
      </div>
      <div>
        Mocked Starter Form
      </div>
    </div>
  </div>
</body>
`;

exports[`PollForm snapshot when poll is not finished renders with description 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <div>
        Mocked Description
      </div>
    </div>
  </div>
</body>
`;
